### PR TITLE
chore(ui-react): lint primitives (T - Z)

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Lint primitives `T*` - `Z*`

#### Issue #, if available
NA

#### Description of how you validated changes
`yarn react lint && yarn react test`
#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/.eslintrc.js [hunk 1]
```diff
@@ -30,13 +30,13 @@ const primitivePatterns = templateJoin(
...
     'R',
     'S',
-    // 'T',
-    // 'U',
-    // 'V',
-    // 'W',
-    // 'X',
-    // 'Y',
-    // 'Z'
+    'T',
```
Comment (wlee221) [inline packages/react/.eslintrc.js:33]: If all primitives are done, can we remove these and go back to `src/primitives/**/*`?

File: packages/react/src/primitives/Tabs/Tabs.tsx [hunk 2]
```diff
@@ -72,20 +76,12 @@ const TabsPrimitive: Primitive<TabsProps, typeof Flex> = (
...
           return (
             <Panel key={index} value={`${index}`}>
               {child.props.children}
             </Panel>
           );
-        } else {
-          // at this point the child defined (not null or undefined)
-          // it is NOT a TabItem, so log a message
-          if (child) {
-            console.warn(
```
Comment (wlee221) [inline packages/react/src/primitives/Tabs/Tabs.tsx:85]: I know you explained this, but why are we not logging a message anymore?

**Review by 0618 — APPROVED** [review]

File: packages/react/src/primitives/Tabs/Tabs.tsx (hunk 2 above)
Comment (calebpollman) [reply packages/react/src/primitives/Tabs/Tabs.tsx:85]: It was very easy to bypass the `isTabsType` predicate, so the warning here was also fragile

File: packages/react/.eslintrc.js (hunk 1 above)
Comment (calebpollman) [reply packages/react/.eslintrc.js:33]: There is no going back where we've never been 😄 

All jokes aside will update the _.eslintrc.js_ in a follow up

File: packages/react/src/primitives/Tabs/Tabs.tsx [hunk 3]
```diff
@@ -9,17 +9,21 @@ import * as React from 'react';
...
 
-const isTabsType = (child: any): child is React.ReactElement<TabItemProps> => {
-  return (
-    child !== null &&
-    typeof child === 'object' &&
-    child.hasOwnProperty('props') &&
-    child.props.title != null
-  );
-};
+// `TabItemProps` does not include `data-apacing` or `value` but they are available
```
Comment (joebuono) [inline packages/react/src/primitives/Tabs/Tabs.tsx:15]: nit: `data-spacing`

**Review by joebuono — DISMISSED** [review]

File: packages/react/.eslintrc.js (hunk 1 above)
Comment (calebpollman) [reply packages/react/.eslintrc.js]: Went ahead and updated the lint config to remove the need `templateJoin`

**Review by wlee221 — APPROVED** [review]

File: packages/react/.eslintrc.js [hunk 4]
```diff
@@ -1,78 +1,27 @@
...
   ignorePatterns: [
     // top level directories
     '__tests__',
     'dist',
-    'node_modules',
-    'scripts',
 
     // config files
+    '*.config.js',
     '.eslintrc.js',
```
Comment (calebpollman) [inline packages/react/.eslintrc.js:11]:
```suggestion
    '.eslintrc.js',

    // TODO remove once the full set of rules can be turned on for the repo
    'src/react-shim.js',
```

File: packages/react/src/primitives/Tabs/Tabs.tsx [hunk 5]
```diff
@@ -9,18 +9,26 @@ import * as React from 'react';
...
-  );
+/**
+ * `TabItemProps` does not include HTML data attributes, so `data-spacing` is added explicitly
+ * to the props of the return type of `isExtendedTabItem` to allow passing of `data-spacing`
+ * to `TabsItem` from inside `Tabs`.
+ * 
+ * Additionally, `value` is avaialble on the props of `TabItemPrimitive`, but is not present
+ * on `TabItemProps`. To mitigate this issue prefer usage of the props of `TabItemPrimitive`
+`*/
+type ExtendedTabItemProps = Parameters<typeof TabItemPrimitive>[0] & {
```
Comment (reesscot) [inline packages/react/src/primitives/Tabs/Tabs.tsx:23]: TIL about `Parameters` util type!

**Review by reesscot — APPROVED** [review]
LGTM

**Review by wlee221 — APPROVED** [review]

**Pull request merged**